src/main/java/org/alliancegenome/curation_api/services/ReferenceService.java

```diff
@@ -56,16 +59,40 @@ public Reference retrieveFromDbOrLiteratureService(String curieOrXref) {
 				reference = referenceCacheMap.get(curieOrXref);
 			} else {
 				Log.debug("Reference not cached, caching reference: (" + curieOrXref + ")");
-				reference = findOrCreateReference(curieOrXref);
-				referenceCacheMap.put(curieOrXref, reference);
+				if (referenceCacheMap.isEmpty()) {
+					referenceCacheMap = referenceDAO.getReferenceMap();
+				} else {
+					reference = findOrCreateReference(curieOrXref);
+					referenceCacheMap.put(curieOrXref, reference);
+				}
 			}
 		} else {
 			reference = findOrCreateReference(curieOrXref);
 			referenceRequest = new Date();
 		}
 		return reference;
 	}
-	
+
+	@Transactional
+	public Reference retrieveShallowReferenceFromDbOrLiteratureService(String pmid) {
+		Reference reference = null;
+		if (shallowReferenceCacheMap.containsKey(pmid)) {
+			reference = shallowReferenceCacheMap.get(pmid);
+		} else {
+			Log.debug("Reference not cached, caching reference: (" + pmid + ")");
+			if (shallowReferenceCacheMap.isEmpty()) {
+				shallowReferenceCacheMap = referenceDAO.getShallowReferenceMap();
+				reference = shallowReferenceCacheMap.get(pmid);
+			} else {
+/*
+					reference = findOrCreateReference(curieOrXref);
```

Comment:
So what happens if the reference isn't in our database?

Reply:
adding the logic to create the reference (from the literature system)